### PR TITLE
feat: Analytics for pinned activities - MEED-413 - meeds-io/meeds#14

### PR DESCRIPTION
**Pull request by GouadriaHanen**
prior to this change, no analytics for Pin and unpin activities are added,
after this change, added analytics for pin operation

### CODE REVIEW

File: analytics-webapps/src/main/webapp/js/statistic-collection.js [hunk 1]
```diff
@@ -70,8 +70,9 @@ function() {
           'name': 'search by favorite',
           'timestamp': Date.now()
         }));
+        document.addEventListener('activity-stream-pinned', event => this.addStatisticActivityPin(true, event && event.detail));
```
Comment (AzmiTouil) [inline analytics-webapps/src/main/webapp/js/statistic-collection.js]:
```suggestion
        document.addEventListener('activity-pinned', event => this.addStatisticActivityPin(true, event && event.detail));
```

File: analytics-webapps/src/main/webapp/js/statistic-collection.js [hunk 2]
```diff
@@ -70,8 +70,9 @@ function() {
...
+        document.addEventListener('activity-stream-unpinned', event => this.addStatisticActivityPin(false, event && event.detail));
```
Comment (AzmiTouil) [inline analytics-webapps/src/main/webapp/js/statistic-collection.js]:
```suggestion
        document.addEventListener('activity-unpinned', event => this.addStatisticActivityPin(false, event && event.detail));
```

File: analytics-webapps/src/main/webapp/js/statistic-collection.js [hunk 3]
```diff
@@ -286,6 +287,23 @@ function() {
...
+    addStatisticActivityPin: function (pinned, eventDetail) {
+      let type = eventDetail.activityType;
+      let spaceId = eventDetail.spaceId || '';
+      this.sendMessage({
+        'module': 'social',
+        'subModule': 'activity',
+        'userId': eXo.env.portal.userIdentityId,
+        'userName': eXo.env.portal.userName,
+        'name': 'pin',
+        'operation': pinned && 'Pin' || 'UnPin',
```
Comment (AzmiTouil) [inline analytics-webapps/src/main/webapp/js/statistic-collection.js:313]: Add translated labels for operations please!

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: analytics-webapps/src/main/webapp/js/statistic-collection.js (hunk 3 above)
Comment (GouadriaHanen) [reply analytics-webapps/src/main/webapp/js/statistic-collection.js:313]: done

File: analytics-webapps/src/main/webapp/js/statistic-collection.js (hunk 2 above)
Comment (GouadriaHanen) [reply analytics-webapps/src/main/webapp/js/statistic-collection.js]: done!

File: analytics-webapps/src/main/webapp/js/statistic-collection.js (hunk 1 above)
Comment (GouadriaHanen) [reply analytics-webapps/src/main/webapp/js/statistic-collection.js]: done!

File: analytics-webapps/src/main/webapp/js/statistic-collection.js [hunk 4]
```diff
@@ -302,6 +303,23 @@ function() {
         cCometd.publish(this.settings.cometdChannel, JSON.stringify(statisticMessage));
       }
     },
+    addStatisticActivityPin: function (pinned, eventDetail) {
+      let type = eventDetail.type;
+      let spaceId = eXo.env.portal.spaceId || eventDetail.activityStream?.space?.dataEntity?.id;
```
Comment (AzmiTouil) [inline analytics-webapps/src/main/webapp/js/statistic-collection.js]:
```suggestion
```

File: analytics-webapps/src/main/webapp/js/statistic-collection.js [hunk 5]
```diff
@@ -302,6 +303,23 @@ function() {
...
+        'module': 'social',
+        'subModule': 'activity',
+        'userId': eXo.env.portal.userIdentityId,
+        'userName': eXo.env.portal.userName,
+        'name': 'pin',
+        'operation': pinned && 'Pin' || 'UnPin',
+        'timestamp': Date.now(),
+        'parameters': {
+          'type': type,
+          'spaceId': spaceId,
```
Comment (AzmiTouil) [inline analytics-webapps/src/main/webapp/js/statistic-collection.js]:
```suggestion
          'type': eventDetail.type,
          'spaceId': eventDetail.activityStream?.space?.dataEntity?.id,
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: analytics-webapps/src/main/webapp/js/statistic-collection.js (hunk 5 above)
Comment (GouadriaHanen) [reply analytics-webapps/src/main/webapp/js/statistic-collection.js]: done

File: analytics-webapps/src/main/webapp/js/statistic-collection.js (hunk 4 above)
Comment (GouadriaHanen) [reply analytics-webapps/src/main/webapp/js/statistic-collection.js]: done

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**